sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/UnwrapCastSuite.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.optimizer
+
+import org.apache.spark.sql.catalyst.dsl.expressions._
+import org.apache.spark.sql.catalyst.dsl.plans.DslLogicalPlan
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.RuleExecutor
+import org.apache.spark.sql.types.DoubleType
+
+class UnwrapCastSuite extends PlanTest {
+
+  object Optimize extends RuleExecutor[LogicalPlan] {
+    val batches: List[Batch] =
+      Batch("Unwrap casts", FixedPoint(10), ConstantFolding, UnwrapCast) :: Nil
+  }
+
+  val testRelation: LocalRelation = LocalRelation('a.short, 'b.float)
+
+  test("unwrap casts when literal == max") {
+    val v = Short.MaxValue
+    assertEquivalent('a > v.toInt, UnwrapCast.falseIfNotNull('a))
+    assertEquivalent('a >= v.toInt, 'a === v)
+    assertEquivalent('a === v.toInt, 'a === v)
+    assertEquivalent('a <=> v.toInt, 'a === v)
+    assertEquivalent('a <= v.toInt, UnwrapCast.trueIfNotNull('a))
+    assertEquivalent('a < v.toInt, 'a =!= v)
+  }
+
+  test("unwrap casts when literal > max") {
+    val v: Int = Short.MaxValue + 100
```

Comment:
Shall we use the predefined one instead of adding a random one?
```scala
+import org.apache.spark.sql.catalyst.expressions.IntegralLiteralTestUtils._
...
-    val v: Int = Short.MaxValue + 100
+    val v: Int = positiveInt
```

Reply:
Didn't know this exist. Will switch.